zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/client/RealmAwareZkClient.java

```diff
@@ -261,8 +261,11 @@ void asyncSetData(final String path, Object datat, final int version,
 
   long getCreationTime(String path);
 
+  @Deprecated
   List<OpResult> multi(final Iterable<Op> ops);
```

Comment:
Is deprecating this is a necessity here? zookeeper-api is supposed to be a general-purpose client library as well (not just tailored for Helix usage), so I don't see the need for it quite yet. My opinion is that we should continue to maintain the interface defined in the original ZkClient.

Reply:
This operation does not fit the Helix ZkClient design since,
1. there is no retry until connecting support.
2. there is no metric support.
Keep supporting it, then we need to take it as a very special case. Currently, there is no large scale usage, so the metrics and caller logic looks fine. Otherwise, we will see issues here and there.
So my take is that we want to replace it with the new multiops API with full support.